### PR TITLE
[KHRGA-37] Updated range classes

### PR DESCRIPTION
**Pull request by imvucic**
Issue: KHRGA-37

### CODE REVIEW

File: source/iface/range.rst [hunk 1]
```diff
@@ -12,19 +12,13 @@
...
-
-
-.. rubric:: Template parameters
-
-================  ===
-``dimensions``    Number of dimensions
-================  ===
+``range<int Dimensions>`` is a 1D, 2D or 3D vector that defines
+the iteration domain of either a single work-group in a parallel
+dispatch, or the overall Dimensions of the dispatch.
```
Comment (mkinsner) [inline source/iface/range.rst:20]:
```suggestion
dispatch, or the overall dimensions of the dispatch.
```

File: source/iface/range.rst [hunk 2]
```diff
@@ -33,58 +27,160 @@ numbers, depending on the dimensionality of the object it describes.
...
+
+Where ``OP`` is: ``+``, ``-``, ``*``, ``/``, ``%``, ``<<``,
+``>>``, ``&``, ``|``, ``^``, ``&&``, ``||``, ``<``, ``>``,
+``<=``, ``>=``.
+
+Constructs and returns a new instance of the SYCL ``range`` class template
+with the same dimensionality as the ``rhs`` SYCL ``range``, where each
+element of the new SYCL ``range`` instance is the result of an element-wise
+``OP`` operator between the ``lhs size_t`` and each element of the
+``rhs`` SYCL ``ange``. If the operator returns a ``bool``,
```
Comment (mkinsner) [inline source/iface/range.rst:154]:
```suggestion
``rhs`` SYCL ``range``. If the operator returns a ``bool``,
```

File: source/iface/range.rst (hunk 1 above)
Comment (imvucic) [reply source/iface/range.rst]: Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/107/commits/da1c44961f750382a7ed4fb16bba2ea9aa1e5492

File: source/iface/range.rst (hunk 2 above)
Comment (imvucic) [reply source/iface/range.rst]: Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/107/commits/da1c44961f750382a7ed4fb16bba2ea9aa1e5492

**Review by mkinsner — APPROVED** [review]

**Pull request merged**